### PR TITLE
Feature/token overlay

### PR DESCRIPTION
**Pull request by Velenir**
Refactoring of TokenOverlay component to make it more modular.

### CODE REVIEW

File: src/styles/components/home/_home.scss [hunk 1]
```diff
@@ -421,7 +421,7 @@
       position: fixed;
     }
 
-    > ul {
+    > .tokenList {
```
Comment (W3stside) [inline src/styles/components/home/_home.scss:424]: You verified this won't conflict somewhere else? Mainly concerned as this is indented CSS styling

File: src/components/TokenList/index.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import TokenItem from '../TokenItem'
+import { code2tokenMap, TokenCode } from 'globals'
+
+
+interface TokenListProps {
+  tokens: TokenCode[],
+  balances: {[code in TokenCode]: number }
```
Comment (W3stside) [inline src/components/TokenList/index.tsx:8]: this is TS typing for an Object within an Object yeah?

File: src/components/TokenOverlay/index.tsx [hunk 3]
```diff
@@ -1,29 +1,30 @@
...
-  GNT: 'GOLEM',
-}
+import TokenOverlayHeader from '../TokenOverlayHeader'
+import TokenList from '../TokenList'
+import { code2tokenMap, TokenCode } from 'globals'
+import { createSelector } from 'reselect'
+
+const filterTokens = createSelector(
+  (state: TokenOverlayState, _: TokenOverlayProps) => state.filter.toUpperCase(),
+  (_: TokenOverlayState, props: TokenOverlayProps) => props.tokenCodeList,
```
Comment (W3stside) [inline src/components/TokenOverlay/index.tsx:9]: what is the '_' used as param 1?

Comment (Velenir) [reply src/components/TokenOverlay/index.tsx:9]: Just a convention meaning `_` param isn't going to be used. The first two arrow function must have the same signature as that is what `filterTokens` will receive as input and pass onto them.

File: src/components/TokenList/index.tsx (hunk 2 above)
Comment (Velenir) [reply src/components/TokenList/index.tsx:8]: balances is an object with keys, that are any of TokenCode values, that is:
```
balances = {
  ETH: 6578,
  GNO: 12345,
  ...
}
```

it is kinda like array comprehension in python

File: src/styles/components/home/_home.scss (hunk 1 above)
Comment (Velenir) [reply src/styles/components/home/_home.scss:424]: So far as token list isn't used anywhere else, it won't conflict

**Review by W3stside — APPROVED** [review]
Yay - looks solid, works well. Good stuff

**Pull request merged**